PluginCore/PluginCore/Interfaces.cs

```diff
@@ -98,79 +98,287 @@ public interface IMainForm : IContainerControl, IWin32Window
+        /// </summary>
         String ProcessArgString(String args);
+        /// <summary>
+        /// Gets the specified item's shortcut keys.
+        /// </summary>
         Keys GetShortcutItemKeys(String id);
+        /// <summary>
+        /// Gets the specified item's id.
+        /// </summary>
         String GetShortcutItemId(Keys keys);
+        /// <summary>
+        /// Gets a theme property value.
+        /// </summary>
         String GetThemeValue(String id);
+        /// <summary>
+        /// Gets a theme property color.
+        /// </summary>
         Color GetThemeColor(String id);
+        /// <summary>
+        /// Gets a theme flag value.
+        /// </summary>
         Boolean GetThemeFlag(String id);
+        /// <summary>
+        /// Gets a theme flag value with a fallback.
+        /// </summary>
         Boolean GetThemeFlag(String id, Boolean fallback);
+        /// <summary>
+        /// Gets a theme property value with a fallback.
+        /// </summary>
         String GetThemeValue(String id, String fallback);
+        /// <summary>
+        /// Gets a theme property color with a fallback.
+        /// </summary>
         Color GetThemeColor(String id, Color fallback);
+        /// <summary>
+        /// Finds the specified plugin.
+        /// </summary>
         IPlugin FindPlugin(String guid);
+        /// <summary>
+        /// Adjusts the image for different themes.
+        /// </summary>
         Image ImageSetAdjust(Image image);
+        /// <summary>
+        /// Finds the specified composed/ready image.
+        /// </summary>
         Image FindImage(String data);
 
         #endregion
 
         #region IMainFrom Properties
 
+        /// <summary>
+        /// Gets the <see cref="ISettings"/> interface.
+        /// </summary>
         ISettings Settings { get; }
+        /// <summary>
+        /// Gets the tool strip.
+        /// </summary>
         ToolStrip ToolStrip { get; }
+        /// <summary>
+        /// Gets the menu strip.
+        /// </summary>
         MenuStrip MenuStrip { get; }
+        /// <summary>
+        /// Gets the <see cref="Scintilla"/> configuration.
+        /// </summary>
         Scintilla SciConfig { get; }
+        /// <summary>
+        /// Gets the dock panel.
+        /// </summary>
         DockPanel DockPanel { get; }
+        /// <summary>
+        /// Gets the application start arguments.
+        /// </summary>
         String[] StartArguments { get; }
+        /// <summary>
+        /// Gets the application custom arguments.
+        /// </summary>
         List<Argument> CustomArguments { get; }
+        /// <summary>
+        /// Gets the status strip.
+        /// </summary>
         StatusStrip StatusStrip { get; }
+        /// <summary>
+        /// Gets or sets the working directory.
+        /// </summary>
         String WorkingDirectory { get; set; }
+        /// <summary>
+        /// Gets the tool strip panel.
```

Comment:
hehehehe i could feel that
originally I was going to add very descriptive documentation, including <param> tags and stuff. After like 5 methods, I was like. "You know what? NO. No just no."